### PR TITLE
Add number tile feature

### PR DESCRIPTION
**Pull request by josephabbey**
## Proposed change

Number Input (buttons or slider)
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
type: tile
entity: input_number.test
features:
  - type: number
    style: buttons
  - type: number
    style: slider
```

## Additional information

- This PR fixes: [fixes #](https://github.com/home-assistant/frontend/pull/18501)

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/tile-features/hui-number-tile-feature.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+              .disabled=${isUnavailableState(stateObj.state)}
+            ></ha-control-number-buttons>`
+          : html`<ha-control-slider
+              value=${stateObj.state}
+              min=${stateObj.attributes.min}
+              max=${stateObj.attributes.max}
+              step=${stateObj.attributes.step}
+              @value-changed=${this._setValue}
+              .disabled=${isUnavailableState(stateObj.state)}
+              >test
```
Comment (piitaya) [inline src/panels/lovelace/tile-features/hui-number-tile-feature.ts:95]:
```suggestion
              >
```

File: src/panels/lovelace/tile-features/hui-number-tile-feature.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import "../../../components/ha-control-button";
+import "../../../components/ha-control-button-group";
+import "../../../components/ha-control-number-buttons";
+import "../../../components/ha-control-slider";
+import "../../../components/ha-icon";
+
+export const supportsNumberTileFeature = (stateObj: HassEntity) => {
+  const domain = computeDomain(stateObj.entity_id);
+  return domain === "input_number";
+};
```
Comment (piitaya) [inline src/panels/lovelace/tile-features/hui-number-tile-feature.ts:18]: It must supports `number` domain too.

File: src/panels/lovelace/tile-features/hui-number-tile-feature.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    }
+  }
+
+  private async _setValue(ev: CustomEvent) {
+    const stateObj = this.stateObj!;
+
+    await this.hass!.callService("input_number", "set_value", {
+      entity_id: stateObj.entity_id,
+      value: ev.detail.value,
+    });
```
Comment (piitaya) [inline src/panels/lovelace/tile-features/hui-number-tile-feature.ts:62]: It must supports `number` domain too.

File: src/translations/en.json [hunk 4]
```diff
@@ -5207,6 +5207,14 @@
                   "select-options": {
                     "label": "Select options"
                   },
+                  "number": {
+                    "label": "Number Input",
```
Comment (piitaya) [inline src/translations/en.json:5211]:
```suggestion
                    "label": "Number",
```
It's not always an input number.

**Review by piitaya — CHANGES_REQUESTED** [review]
Thanks ! I added few comments.

File: src/panels/lovelace/tile-features/hui-number-tile-feature.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    super.willUpdate(changedProp);
+    if (changedProp.has("stateObj") && this.stateObj) {
+      this._currentState = this.stateObj.state;
+    }
+  }
+
+  private async _setValue(ev: CustomEvent) {
+    const stateObj = this.stateObj!;
+
+    await this.hass!.callService("input_number", "set_value", {
```
Comment (Misiu) [inline src/panels/lovelace/tile-features/hui-number-tile-feature.ts:59]: I think that for number domain the service that should be called is `number.set_value`

Comment (JosephAbbey) [reply src/panels/lovelace/tile-features/hui-number-tile-feature.ts:59]: Yes, I just added that domain so I hadn't tested it yet, good catch.

File: src/panels/lovelace/editor/config-elements/hui-tile-card-features-editor.ts [hunk 6]
```diff
@@ -90,6 +93,7 @@ const SUPPORTS_FEATURE_TYPES: Record<
   "vacuum-commands": supportsVacuumCommandTileFeature,
   "water-heater-operation-modes": supportsWaterHeaterOperationModesTileFeature,
   "select-options": supportsSelectOptionTileFeature,
+  number: supportsNumberTileFeature,
```
Comment (Misiu) [inline src/panels/lovelace/editor/config-elements/hui-tile-card-features-editor.ts:96]:
```suggestion
  "number": supportsNumberTileFeature,
```
this should be added for consistency

File: src/panels/lovelace/tile-features/hui-number-tile-feature.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    super.willUpdate(changedProp);
+    if (changedProp.has("stateObj") && this.stateObj) {
+      this._currentState = this.stateObj.state;
+    }
+  }
+
+  private async _setValue(ev: CustomEvent) {
+    const stateObj = this.stateObj!;
+
+    const domain = computeDomain(stateObj.entity_id);
```
Comment (Misiu) [inline src/panels/lovelace/tile-features/hui-number-tile-feature.ts:59]: can we computeDomain once or memoize it? Now it is calculated every time we want to change the value.

File: src/panels/lovelace/editor/config-elements/hui-tile-card-features-editor.ts (hunk 6 above)
Comment (piitaya) [reply src/panels/lovelace/editor/config-elements/hui-tile-card-features-editor.ts:96]: Formatting is automatically done by prettier.

File: src/panels/lovelace/tile-features/hui-number-tile-feature.ts (hunk 7 above)
Comment (JosephAbbey) [reply src/panels/lovelace/tile-features/hui-number-tile-feature.ts:59]: This function also only runs on change not on render so it won't run that often.

**Comment by piitaya** [comment]
@JosephAbbey is this ready to review?

**Review by piitaya — APPROVED** [review]

**Comment by piitaya** [comment]
Thank you ! Ok for me !

**Pull request merged**